src/libs/Network/SequentialQueue.js

```diff
@@ -114,9 +114,21 @@ function getCurrentRequest() {
     return currentRequest;
 }
 
+/**
+ * Returns a promise that resolves when the last request on the queue responds.
+ * @returns {Promise}
+ */
+function getIsReadyPromise() {
```

Comment:
I don't really like the name of this method because when thinking about it from an external perspective (ie. outside of this file), I don't think it makes sense `SequentialQueue.getIsReadyPromise()`. What does "is ready" mean and why should anything care that it "is ready"? I'm also not totally sure that the name makes sense from an internal perspective either.

Some ideas for a different name:
- isDone
- isIdle
- hasNoRequests
- hasFinishedAllRequests

I'm sure @marcaaron will have some good ideas for names too.

Reply:
Yeah, I had thought of using `wait` in the name too, so I would be down for that.

Reply:
I agree, I like the `wait` syntax, but to me the word `idle` isn't super clear, I think I prefer something like `waitForComplete`.